modules/raft/src/main/java/org/apache/ignite/internal/raft/RaftGroupServiceImpl.java

```diff
@@ -811,9 +810,20 @@ private CompletableFuture<ClusterNode> resolvePeer(Peer peer) {
         ClusterNode node = cluster.topologyService().getByConsistentId(peer.consistentId());
 
         if (node == null) {
-            return CompletableFuture.failedFuture(new ConnectException("Peer " + peer.consistentId() + " is unavailable"));
+            return CompletableFuture.failedFuture(new PeerUnavailableException("Peer " + peer.consistentId() + " is unavailable"));
```

Comment:
- Please fix the recoverable() javadoc. Currently it states  
`Checks if an error is recoverable, for example, {@link java.net.ConnectException}.` but ConnectException itself isn't considered as recoverable. BWT is a bit funny that ConnectionException wasn't handled.

- There are at least two placed where we still use ConnectException:

1. RaftGroupServiceTest
```
            if (peer != null && target.name().equals(peer.consistentId())) {
                return failedFuture(new ConnectException());
            }
```
2. org.apache.ignite.raft.jraft.rpc.impl.IgniteRpcClient#send
```
        if (targetNode == null) {
            // ConnectException will force a retry by the enclosing components.
            fut.completeExceptionally(new ConnectException());

            return;
        }
```

Reply:
> BWT is a bit funny that ConnectionException wasn't handled.

It was, `ConnectException` is an `IOException`

Reply:
> It was, ConnectException is an IOException
Yep, you are right.

What about other topics?

Reply:
> There are at least two placed where we still use ConnectException:

The first place that you mentioned (`RaftGroupServiceTest`) is a test where we mock the `ClusterService` behavior, I think that using a `ConnectException` is the correct choice there.

Reply:
Ok, and what about `org.apache.ignite.raft.jraft.rpc.impl.IgniteRpcClient#send`?

Reply:
I'm trying to fix it now)

Reply:
I'm not sure about the other place too. We can change the exception type, but then `IgniteRpcClient#send` will be able to throw both `PeerUnavailableException` (when the target is not in the physical topology) and `ConnectException` (when the target is still in the topology, but failed to respond for some reason?). This means that the code inside `AbstractClientService#invokeWithDone` will need to handle both of these exception types. I don't know why it was written like that (to handle `ConnectException` is a special way), but it is what it is

Reply:
Well, I believe that it's better to be consistent. If we consider node == null as PeerUnavailableException we should do the same in IgniteRpcClient#send in case of targetNode == null. Up to you.

Reply:
Fixed

Reply:
Thanks! Proposed changes looks good.